encoding/encodeInto.any.js

```diff
@@ -73,77 +73,91 @@
       "destinationOffset": 4,
       "filler": "random"
     }
-  ].forEach(destinationData => {
-    test(() => {
-      // Setup
-      const bufferLength = testData.destinationLength + destinationData.bufferIncrease,
-            destinationOffset = destinationData.destinationOffset,
-            destinationLength = testData.destinationLength,
-            destinationFiller = destinationData.filler,
-            encoder = new TextEncoder(),
-            buffer = new ArrayBuffer(bufferLength),
-            view = new Uint8Array(buffer, destinationOffset, destinationLength),
-            fullView = new Uint8Array(buffer),
-            control = new Array(bufferLength);
-      let byte = destinationFiller;
-      for (let i = 0; i < bufferLength; i++) {
-        if (destinationFiller === "random") {
-          byte = Math.floor(Math.random() * 256);
+  ].forEach((destinationData) => {
+    [ArrayBuffer, self["SharedArrayBuffer"]].forEach((ArrayBufferOrSharedArrayBuffer) => {
```

Comment:
I guess this works, but it seems slightly better to use `"ArrayBuffer"` and `"SharedArrayBuffer"` here and then use `new self[ArrayBufferOrSharedArrayBuffer]` inside the test. That way there's not a bunch of undefined weirdness and instead the exception will make some sense.